gradle-plugin/src/gradleTest/java/com/squareup/anvil/plugin/IncrementalTest.kt

```diff
@@ -0,0 +1,545 @@
+          """
+          package com.squareup.test
+      
+          import javax.inject.Inject
+          
+          class OtherClass @Inject constructor()
+          """.trimIndent(),
+        )
+      }
+      gradlePropertiesFile(
+        """
+        com.squareup.anvil.trackSourceFiles=true
+        """.trimIndent(),
+      )
+    }
+
+    shouldSucceed("compileJava")
+
+    rootAnvilMainGenerated.injectClassFactory.shouldExist()
+    rootAnvilMainGenerated.resolve("com/squareup/test/OtherClass_Factory.kt").shouldExist()
+
+    workingDir.resolve("src/main/java")
+      .resolve(injectClassPath)
+      .delete()
+
+    shouldSucceed("compileJava") {
+      task(":compileKotlin")?.outcome shouldBe TaskOutcome.SUCCESS
+    }
+
+    rootAnvilMainGenerated.injectClassFactory.shouldNotExist()
+    rootAnvilMainGenerated.resolve("com/squareup/test/OtherClass_Factory.kt").shouldExist()
+  }
+
+  @TestFactory
+  fun `a generated factory is deleted if the @Inject annotation is removed`() =
+    withTrackSourceFiles { (trackSourceFiles) ->
+
+      val injectClassPath = "com/squareup/test/InjectClass.kt"
+
+      rootProject {
+        dir("src/main/java") {
+
+          kotlinFile(
+            injectClassPath,
+            """
+            package com.squareup.test
+      
+            import javax.inject.Inject
+
+            class InjectClass @Inject constructor()
+            """.trimIndent(),
+          )
+        }
+        gradlePropertiesFile(
+          """
+          com.squareup.anvil.trackSourceFiles=$trackSourceFiles
+          """.trimIndent(),
+        )
+      }
+
+      shouldSucceed("compileJava")
+
+      rootAnvilMainGenerated.injectClassFactory.shouldExist()
+
+      rootProject {
+        dir("src/main/java") {
+          kotlinFile(
+            injectClassPath,
+            """
+            package com.squareup.test
+     
+            class InjectClass
+            """.trimIndent(),
+          )
+        }
+      }
+
+      shouldSucceed("compileJava") {
+        task(":compileKotlin")?.outcome shouldBe TaskOutcome.SUCCESS
+      }
+
+      rootAnvilMainGenerated.injectClassFactory.shouldNotExist()
+    }
+
+  @TestFactory
+  fun `a generated factory is updated if the source class constructor is changed`() =
+    testFactory {
+
+      val injectClassPath = "com/squareup/test/InjectClass.kt"
+
+      rootProject {
+        dir("src/main/java") {
+
+          kotlinFile(
+            injectClassPath,
+            """
+              package com.squareup.test
+            
+              import javax.inject.Inject
+                
+              class InjectClass @Inject constructor()
+            """.trimIndent(),
+          )
+        }
+        gradlePropertiesFile(
+          """
+            com.squareup.anvil.trackSourceFiles=true
+          """.trimIndent(),
+        )
+      }
+
+      shouldSucceed("compileJava")
+
+      rootAnvilMainGenerated.injectClassFactory.shouldExist()
```

Comment:
Could you add a factory content assertion here too? I think the explicitness would be valuable since we're focused on the generated content in this case

Reply:
added